app/celery/broadcast_message_tasks.py

```diff
@@ -50,35 +133,56 @@ def send_broadcast_provider_message(broadcast_event_id, provider):
+    try:
+        if broadcast_event.message_type == BroadcastEventMessageType.ALERT:
+            cbc_proxy_provider_client.create_and_send_broadcast(
+                identifier=str(broadcast_provider_message.id),
+                message_number=formatted_message_number,
+                headline="GOV.UK Notify Broadcast",
+                description=broadcast_event.transmitted_content['body'],
+                areas=areas,
+                sent=broadcast_event.sent_at_as_cap_datetime_string,
+                expires=broadcast_event.transmitted_finishes_at_as_cap_datetime_string,
+            )
+        elif broadcast_event.message_type == BroadcastEventMessageType.UPDATE:
+            cbc_proxy_provider_client.update_and_send_broadcast(
+                identifier=str(broadcast_provider_message.id),
+                message_number=formatted_message_number,
+                headline="GOV.UK Notify Broadcast",
+                description=broadcast_event.transmitted_content['body'],
+                areas=areas,
+                previous_provider_messages=broadcast_event.get_earlier_provider_messages(provider),
+                sent=broadcast_event.sent_at_as_cap_datetime_string,
+                expires=broadcast_event.transmitted_finishes_at_as_cap_datetime_string,
+            )
+        elif broadcast_event.message_type == BroadcastEventMessageType.CANCEL:
+            cbc_proxy_provider_client.cancel_broadcast(
+                identifier=str(broadcast_provider_message.id),
+                message_number=formatted_message_number,
+                previous_provider_messages=broadcast_event.get_earlier_provider_messages(provider),
+                sent=broadcast_event.sent_at_as_cap_datetime_string,
+            )
+    except CBCProxyRetryableException as exc:
+        delay = get_retry_delay(self.request.retries)
+        current_app.logger.exception(
+            f'Retrying send_broadcast_provider_message for broadcast_event {broadcast_event_id} and ' +
+            f'provider {provider} in {delay} seconds'
         )
-    elif broadcast_event.message_type == BroadcastEventMessageType.UPDATE:
-        cbc_proxy_provider_client.update_and_send_broadcast(
-            identifier=str(broadcast_provider_message.id),
-            message_number=formatted_message_number,
-            headline="GOV.UK Notify Broadcast",
-            description=broadcast_event.transmitted_content['body'],
-            areas=areas,
-            previous_provider_messages=broadcast_event.get_earlier_provider_messages(provider),
-            sent=broadcast_event.sent_at_as_cap_datetime_string,
-            expires=broadcast_event.transmitted_finishes_at_as_cap_datetime_string,
+
+        # TODO: Decide whether to set to TECHNICAL_FAILURE or ERROR based on response codes from cbc proxy
+        update_broadcast_provider_message_status(
+            broadcast_provider_message,
+            status=BroadcastProviderMessageStatus.TECHNICAL_FAILURE
```

Comment:
If for some reason it failed to call out to the CBC with the alert and raises a retryable exception, why would we update the broadcast provider message status? Surely it's still in a status of 'sending' until we decide to give up and no longer retry (at which point it gets marked as a failure state of some sort)?

Reply:
yeah possibly. I kinda wanted to differentiate "Currently sending right now" from "Sitting in a queue waiting to go again".

Reply:
I'd be very keen to leave it in a sending type state rather then technical failure as this is a completely different mental model to how all our other notification types work. Stuff stays in sending until we know it's been received correctly or decided that it definitely can't be received.

I think we will be able to tell something is sat in a queue because it's still in sending after more than a minute or two and we've probably had a p1 go off about it or something?

Reply:
yeah ive thought about it and i agree with you - if we do this we can also check for tech failure in the check at the top of the code, which means we've got a killswitch if we need to get something out of the retry loop for whatever reason - we can set its status to tech failure manually